mxcubeweb/__init__.py

```diff
@@ -124,9 +128,25 @@ def build_server_and_config(test=False, argv=None):
         config_path = HWR.get_hardware_repository().find_in_repository("mxcube-web")
 
         cfg = Config(config_path)
-
+        db_path = (
+            Path(cfg.flask.USER_DB_PATH)
+            if cfg.flask.USER_DB_PATH
+            else Path(FlaskConfigModel.USER_DB_PATH.default)
+        )
+        db_path.parent.mkdir(parents=True, exist_ok=True, mode=0o700)
```

Comment:
Why do we need to set specific permission bits here?
What is wrong with the defaults?
At the very least we should document if there is a specific reason...

```suggestion
        db_path.parent.mkdir(parents=True, exist_ok=True)
```

Reply:
It's just that database can contain sensitive information and I wanted something really secure in case with only owner rights but if defaults permissions are fine we can change it.

Reply:
I think you are on the right track @Phantominumm. 

There is cause for concern the normal 644 rights would be too permissive for this sort of information. I think 600 would be enough but its possible that execute is needed for some reason.  

We would not in general want either log files or any session database to be readable by any other user than the one that runs the MXCuBE process. Omitting the `mode` will probably make the warning/error disappear from bandit or Sonar Qube but it will still be an issue.

I think it would be quite good if we could keep the database permissions restrictive to i.e 700 or 600. I can understand that having the log files being readable by another user might be practical but there is no real reason for why the database should have any other permissions than 600, right ?

Reply:
Yes i used 700 to give all the rights but there would be no reason to execute from this directory, i think 600 is more than sufficient.

Reply:
I see the concern of course, but again it feels to me like it is not the correct way of solving this (I might be wrong, I am not a security person).

In this case, I do not think we (MAX IV) care if other Linux user accounts have access to this file or not, so I guess we are fine with setting the permission bits to `600` (or `700`).

Reply:
The other way would be provide a separate script for setting up the log files and the databases and then its up to each one how to handle the permissions. One just need to make sure they exist before launching the application. It would make the setup/install of the application more complicated.

I personally think that handling this within the application, **if the files does not already exist** is a quite nice thing. It makes everything much easier to setup. What we want to make sure is that that functionality is as secure as we can make it. It still leaves it up to each one to configure/setup the files separately by creating the files before the application is started.

Reply:
One could imagine a separate switch at the command line that makes the behavior/creation more explicit

Reply:
My take on what a good way to solve this would be: create the containing directory in advance with the correct permission bits so that only allowed user accounts can access the directory and the DB file that will be created in that directory. Straightforward, no need to add complex code in MXCuBE, every facility can set the permissions they want.

Reply:
Ok, with the proposed logic you should be free to do create the directories with the permissions you like. The permissions are only applied on created folders.

However leaving the `mode` argument out would default the mode to 777 (its default mode of `Path.mkdir` so we should probably not do that.

As we see it, the current logic provides a reasonable, more secure and functional solution that does not require that any folders are created before hand.